### PR TITLE
Storage manager implementation

### LINKED ISSUE #1301: Combine storage-related code
The code for storing and retrieving assets from disk, caching them in memory, and tracking modified assets is somewhat scattered. We should combine this code to make it easier to understand, easier to switch storage backends, and easier to improve, such as through parallelization. The first step in this direction is to create a storage manager that manages reading, writing, and caching.

### PR DESCRIPTION
**Pull request by tst-lsavoie**
Initial effort at a storage manager that is responsible for everything related to asset storage, including reading/writing XML, caching, and tracking dirty assets.

The eventual goal is that the storage manager will become an easy place to implement locking so that we can run builds in parallel.

Fixes #1301.

### CODE REVIEW

File: earth_enterprise/src/fusion/autoingest/Asset.h [hunk 1]
```diff
@@ -36,27 +36,27 @@
  ***  ... = asset->config.layers.size();
  ***
  ******************************************************************************/
-class AssetImpl : public khRefCounter, public AssetStorage {
+class AssetImpl : public khRefCounter, public AssetStorage, public StorageManaged {
   friend class AssetHandle_<AssetImpl>;
 
   // private and unimplemented -- illegal to copy an AssetImpl
   AssetImpl(const AssetImpl&);
   AssetImpl& operator=(const AssetImpl&);
```
Comment (tst-basilhuffman) [inline earth_enterprise/src/fusion/autoingest/Asset.h]: maybe better to delete, along with moves also

File: earth_enterprise/src/fusion/autoingest/AssetVersion.h [hunk 2]
```diff
@@ -38,42 +39,41 @@
...
  ***  }
  ******************************************************************************/
-class AssetVersionImpl : public khRefCounter, public AssetVersionStorage {
+class AssetVersionImpl : public khRefCounter, public AssetVersionStorage, public StorageManaged {
   friend class AssetImpl;
   friend class AssetHandle_<AssetVersionImpl>;
 
   // private and unimplemented -- illegal to copy an AssetVersionImpl
   AssetVersionImpl(const AssetVersionImpl&);
   AssetVersionImpl& operator=(const AssetVersionImpl&);
```
Comment (tst-basilhuffman) [inline earth_enterprise/src/fusion/autoingest/AssetVersion.h]: Same comment as above

**Comment by tst-jkent** [comment]
Reviewed and approved.

File: earth_enterprise/src/fusion/autoingest/Asset.h (hunk 1 above)
Comment (tst-lsavoie) [reply earth_enterprise/src/fusion/autoingest/Asset.h]: Good catch. Fixed.

File: earth_enterprise/src/fusion/autoingest/AssetVersion.h (hunk 2 above)
Comment (tst-lsavoie) [reply earth_enterprise/src/fusion/autoingest/AssetVersion.h]: Fixed.

**Comment by tst-basilhuffman** [comment]
Approved

**Pull request merged**